activesupport/lib/active_support/cache.rb

```diff
@@ -567,14 +568,34 @@ def merged_options(call_options)
           end
         end
 
-        # Prefixes a key with the namespace. Namespace and key will be delimited
-        # with a colon.
-        def normalize_key(key, options)
-          key = Cache.expand_cache_key(key)
-          namespace = options[:namespace] if options
-          prefix = namespace.is_a?(Proc) ? namespace.call : namespace
-          key = "#{prefix}:#{key}" if prefix
-          key
+        # Expands and namespaces the cache key. May be overridden by
+        # cache stores to do additional normalization.
+        def normalize_key(key, options = nil)
+          namespaced_key Cache.expand_cache_key(key), options
+        end
+
+        # Prefix the key with a namespace string:
+        #
+        #   namespaced_key 'foo', namespace: 'cache'
+        #   # => 'cache:foo'
+        #
+        # With a namespace block:
+        #
+        #   namespaced_key 'foo', namespace: -> { 'cache' }
+        #   # => 'cache:foo'
+        def namespaced_key(key, options = nil)
+          options = merged_options(options)
+          namespace = options[:namespace]
+
+          if namespace.respond_to?(:call)
+            namespace = namespace.call
+          end
+
+          if namespace
+            "#{namespace}:#{key}"
+          else
+            key
+          end
         end
```

Comment:
Separated this method from `normalize_key` so they may be overridden and called independently by subclasses.

The Redis cache store hashes cache keys larger than 1kB and it namespaces `deleted_matched` globs, which aren't keys to be normalized.

Reply:
This method was removed from public API in Rails 5.0. https://github.com/rails/rails/pull/22215 for context.

Reply:
I'll call it `namespace_key` instead, matching verb tense with `normalize_key`. (It's a different method than the old `namespaced_key` which expanded, sanitized, namespaced, etc.)

Reply:
:+1: